api/src/main/java/javax/enterprise/inject/spi/BeanManager.java

```diff
@@ -406,14 +408,25 @@
      * @throws ContextNotActiveException if there is no active context object for the given scope
      * @throws IllegalArgumentException if there is more than one active context object for the given scope
      */
-    public Context getContext(Class<? extends Annotation> scopeType);
+    Context getContext(Class<? extends Annotation> scopeType);
+
+    /**
+     * Creates a new {@link ManageableContext} that represents the given built-in scope.  This ManageableContext is not
+     * yet active, it must be activated when ready to use.  Upon activation, you may use it to retrieve beans from
+     * the context.
+     *
+     * @param scopeType the  {@linkplain javax.enterprise.context scope}
+     * @return the {@linkplain ManageableContext context object}
+     * @throws IllegalArgumentException if the given scopeType is not a valid built in scope.
+     */
+    ManageableContext createManageableContext(Class<? extends Annotation> scopeType);
```

Comment:
I don't say it won't help them. I say that one of my use case is the 3rd party fw dev that want to change activation of @RequestScoped beans without recreating his own request context and use the built-in one (i.e WebSocket spec example). I'm not sure this feature will allow that but I may be wrong...